xarray/core/iterators.py

```diff
@@ -60,15 +63,14 @@ def __next__(self) -> Iterator[Tree]:
 
     @staticmethod
     @abstractmethod
-    def _iter(children: List[Tree], filter_, stop, maxlevel) -> Iterator[Tree]:
-        ...
+    def _iter(children: list[Tree], filter_, stop, maxlevel) -> Iterator[Tree]: ...
```

Comment:
Add typing to the rest of the arguments? They should probably have the same typing as above right?

Reply:
Good call. Done in: [dc8ad9](https://github.com/pydata/xarray/pull/8879/commits/dc8ad955173effde954f47c3b380fc5bc4a7432c)